tests/helper.py

```diff
@@ -23,3 +23,14 @@ def _validate_custom_gateway_process(port, text, expected):
 
     resp = requests.get(f'http://127.0.0.1:{port}/stream?text={text}').json()
     assert resp == expected
+
+# set_pod_parser returns a parser for worker runtime, which expects list of ports (because external executors
+# can provide multiple ports and hosts). However this parser is not compatible with ContainerPod, Pod and worker runtime.
+# Should we add a seperate parser for Pod?
```

Comment:
@JoanFM you might need to check this
In my opinion, this is fine for now. We don't expect people to use the pod parser on the Worker runtime or Pod anyways

Reply:
what does it mean is not compatible wirh ContainerPod why not?

Reply:
because set_pod_parser generates args where port and host are lists.
This works fine with Flow and Deployment.
In the deployment layer, the lists are transformed to single values (for each shard and replica). The Pod and Worker runtime therefore expect single values. That's why, set_pod_parser cannot be used with Pod and worker runtime